src/rgw/rgw_op.cc

```diff
@@ -5652,7 +5651,7 @@ void RGWCompleteMultipart::execute()
     return;
   }
 
-  if (!data || !len) {
+  if (!data.c_str() || !data.length()) {
```

Comment:
I suspect we do not want to call c_str() here, it's basically equivalent to data.length() != 0, and we're creating and discarding a buffer

Reply:
Ok

Reply:
Definitely don't call c_str here. Avoid it on buffer::list if at all possible.

For buffer::ptr, its the correct (if badly named) alternative.